### PR TITLE
Add dropout for conditioning cross-attention blocks.

### PR DESCRIPTION
**Pull request by virginiafdez**
Added "dropout_cattn" parameter to add dropout on the conditioning cross-attention channels.
Does not apply dropout everywhere, just in the conditioning blocks.

### CODE REVIEW

**Review by marksgraham — CHANGES_REQUESTED** [review]
This looks good to me - but I think the test case should just be included with the others in the 'test_diffusion_model_unet` file!

**Comment by virginiafdez** [comment]
@marksgraham  Done!

File: tests/test_diffusion_model_unet.py [hunk 1]
```diff
@@ -524,6 +577,17 @@ def test_script_conditioned_3d_models(self):
...
 
+    # Test dropout specification for cross-attention blocks
+    @parameterized.expand(DROPOUT_WRONG)
+    def test_wrong_dropout(self, input_param):
+        with self.assertRaises(ValueError):
+            net = DiffusionModelUNet(**input_param)
+
+    @parameterized.expand(DROPOUT_OK)
+    def test_right_dropout(self, input_param):
+        net = DiffusionModelUNet(**input_param)
```
Comment (marksgraham) [inline tests/test_diffusion_model_unet.py:588]:
```suggestion
        _ = DiffusionModelUNet(**input_param)
```

File: tests/test_diffusion_model_unet.py [hunk 2]
```diff
@@ -524,6 +577,17 @@ def test_script_conditioned_3d_models(self):
             net, torch.rand((1, 1, 16, 16, 16)), torch.randint(0, 1000, (1,)).long(), torch.rand((1, 1, 3))
         )
 
+    # Test dropout specification for cross-attention blocks
+    @parameterized.expand(DROPOUT_WRONG)
+    def test_wrong_dropout(self, input_param):
+        with self.assertRaises(ValueError):
+            net = DiffusionModelUNet(**input_param)
```
Comment (marksgraham) [inline tests/test_diffusion_model_unet.py:584]:
```suggestion
            _ = DiffusionModelUNet(**input_param)
```

**Pull request merged**